File: nysol_python/mcmd/methods/convTex/db/smp/mcommon.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mcommon'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,quantity
A,1
B,2
C,1
D,3
E,1
'''
db['idatas'].append(data)

data={}
data['name']='ref1.csv'
data['text']='''
customer,gender
A,female
B,male
E,female
'''
db['idatas'].append(data)

data={}
data['name']='ref2.csv'
data['text']='''
customerID,gender
A,female
B,male
E,female
'''
db['idatas'].append(data)

data={}
data['name']='dat3.csv'
data['text']='''
customer,quantity
A,1
A,2
A,3
B,1
D,1
D,2
'''
db['idatas'].append(data)

data={}
data['name']='ref3.csv'
data['text']='''
customer
A
A
D
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
入力ファイルにある ``customer`` 項目と、参照ファイルにある ``customer`` 項目が同じ値を持つ入力ファイルの行を選択する。
それ以外のデータは ``oth.csv`` に出力する。
'''
script['sh_code']='''
mcommon k=顧客 m=ref1.csv u=oth.csv i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcommon(k="customer", m="ref1.csv", u="oth.csv", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='oth.csv,rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='同じ値を持たない入力ファイルの行選択'
script['comment']='''
``r=True`` オプションを付けることで、条件が逆転し、参照ファイルにない ``customer`` を選択することになる。
'''
script['sh_code']='''
mcommon k=顧客 m=ref1.csv -r i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcommon(k="customer", m="ref1.csv", r=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='結合キー項目名が異なる場合'
script['comment']='''
結合キーの項目名が異なる場合は、 ``K=`` で指定する。
'''
script['sh_code']='''
mcommon k=顧客 K=顧客ID i=dat1.csv m=ref2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcommon(k="customer", K="customerID", i="dat1.csv", m="ref2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='キー項目に重複行がある場合の例'
script['comment']='''
参照ファイルと入力ファイルのキー項目に重複行があっても選択可能。
'''
script['sh_code']='''
mcommon k=顧客 m=ref3.csv -r i=dat3.csv o=rsl4.csv
'''
script['py_code']='''
nm.mcommon(k="customer", m="ref3.csv", r=True, i="dat3.csv", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mcommon.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Quantity
A,1
B,2
C,1
D,3
E,1
EOF
)}

File.open("ref1.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Gender
A,Female
B,Male
E,Female
EOF
)}

File.open("ref2.csv","w"){|fpw| fpw.write(
<<'EOF'
CustomerID,Gender
A,Female
B,Male
E,Female
EOF
)}

File.open("dat3.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Quantity 
A,1
A,2
A,3
B,1
D,1
D,2
EOF
)}

File.open("ref3.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer
A
A
D
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Select records with common customers in both input file and reference file. Save unmatched records to a separate file \verb|oth.csv|.
EOF
scp=<<'EOF'
more dat1.csv
more ref1.csv
mcommon k=Customer m=ref1.csv u=oth.csv i=dat1.csv o=rsl1.csv
more rsl1.csv
more oth.csv
EOF
run(scp,title,comment)

############## Example 2
title="Select unmatched records from the input file"
comment=<<'EOF'
Reverse selection criteria by using the \verb|-r| option, the "Customer" not in the reference file are selected.
EOF
scp=<<'EOF'
mcommon k=Customer m=ref1.csv -r i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Different names of join key"
comment=<<'EOF'
If the join key field name in the reference file is different, specify the field name at ¥verb|K=|.
EOF
scp=<<'EOF'
more ref2.csv
mcommon k=Customer K=CustomerID i=dat1.csv m=ref2.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

############## Example 4
title="Example with duplicate key fields"
comment=<<'EOF'
Record selection with duplicate records in both input file and reference file.
EOF
scp=<<'EOF'
more dat3.csv
more ref3.csv
mcommon k=Customer m=ref3.csv -r i=dat3.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)
